### PR TITLE
feat: generate protobuf go models 

### PR DESCRIPTION
**Pull request by kruskall**
Add POSIX shell scripts to install protobuf and generate go models: protobuf is downloaded to the `/tmp` directory and installed in `tools/bin` along with `protoc-gen-go`. Go models are generated in the `model/modelpb` directory.

Make tasks are added for ease of use.

Related to https://github.com/elastic/apm-data/issues/36

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,7 +2,10 @@ test:
 	go test -v ./...
 
 fmt:
-	go run golang.org/x/tools/cmd/goimports@v0.3.0 -w .
+	go run golang.org/x/tools/cmd/goimports@v0.3.0 -w ./input
+	go run golang.org/x/tools/cmd/goimports@v0.3.0 -w ./model/modelprocessor
+	go run golang.org/x/tools/cmd/goimports@v0.3.0 -w ./model/internal
+	go run golang.org/x/tools/cmd/goimports@v0.3.0 -w ./model/*.go
```
Comment (axw) [inline Makefile:8]: What's the reason for not formatting the generated code as well?

**Review by axw — COMMENTED** [review]
Looks good. I think we should also use [vtprotobuf](https://github.com/planetscale/vtprotobuf), like at https://github.com/elastic/apm-ingest-service/blob/a0bcd8e2c2039471672383c5ba1c6f7ddc6bf5a4/Makefile#L42-L72

File: Makefile (hunk 1 above)
Comment (kruskall) [reply Makefile:8]: `goimports` is putting stdlib import first which is conflicting with the output generated from protobuf.

To give an example with `os.pb.go`, here's a diff of the file after generation and after `goimports` was applied:

```patch
diff --git a/model/modelpb/os.pb.go b/model/modelpb/os.pb.go
index 021e22f..25f0ce6 100644
--- a/model/modelpb/os.pb.go
+++ b/model/modelpb/os.pb.go
@@ -24,10 +24,11 @@
 package modelpb
 
 import (
-       protoreflect "google.golang.org/protobuf/reflect/protoreflect"
-       protoimpl "google.golang.org/protobuf/runtime/protoimpl"
        reflect "reflect"
        sync "sync"
+
+       protoreflect "google.golang.org/protobuf/reflect/protoreflect"
+       protoimpl "google.golang.org/protobuf/runtime/protoimpl"
 )
 
 const (
```

Comment (axw) [reply Makefile:8]: Why is that a problem? It should be fine to reorder to goimports style.

File: tools/generate-modelpb.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+TOOLS_DIR=$(dirname "$(readlink -f -- "$0")")
+
+PATH="${PATH}:${TOOLS_DIR}/bin" protoc --proto_path=./model/proto/ --go_out=. --go_opt=module=github.com/elastic/apm-data --go-vtproto_out=. --go-vtproto_opt=features=marshal+unmarshal+size,module=github.com/elastic/apm-data ./model/proto/*.proto
```
Comment (axw) [inline tools/generate-modelpb.sh:5]:
```suggestion
PATH="${PATH}:${TOOLS_DIR}/bin" protoc --proto_path=./model/proto/ --go_out=. --go_opt=module=github.com/elastic/apm-data --go-vtproto_out=. --go-vtproto_opt=features=marshal+unmarshal+size+pool,module=github.com/elastic/apm-data ./model/proto/*.proto
```

**Review by axw — APPROVED** [review]
LGTM! I think we want to generate pooling support for vtproto as well.

**Pull request merged**